### PR TITLE
Add support to replace amq with http

### PR DESCRIPTION
**Pull request by aneeshkp**
This PR adds subscription object to store client connections and subscription details

Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>

### CODE REVIEW

File: pkg/store/subscriber/subscriber.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"github.com/redhat-cne/sdk-go/pkg/subscriber"
+	"sync"
+)
+
+// Store  defines subscribers connection store struct
+type Store struct {
+	sync.RWMutex
+	// PublisherStore stores publishers in a map
```
Comment (jzding) [inline pkg/store/subscriber/subscriber.go:11]: Store stores subscribers in a map

File: pkg/subscriber/subscriber.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+// Subscriber object holds client connections
+type Subscriber struct {
+	// ClientID of the sub
+	// +required
+	ClientID string `json:"clientID" omit:"empty"`
+	// +required
+	SubStore *store.PubSubStore `json:"subStore" omit:"empty"`
+	// HealthEndPoint - A URI describing the
+	//producer/subscription get link.
```
Comment (jzding) [inline pkg/subscriber/subscriber.go:44]: producer -> subscriber

File: pkg/store/subscriber/subscriber.go (hunk 1 above)
Comment (aneeshkp) [reply pkg/store/subscriber/subscriber.go:11]: Done

File: pkg/subscriber/subscriber.go (hunk 2 above)
Comment (aneeshkp) [reply pkg/subscriber/subscriber.go:45]: done

File: pkg/protocol/http/http.go [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+	"github.com/redhat-cne/sdk-go/pkg/protocol"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/subscriber"
+	"github.com/redhat-cne/sdk-go/pkg/util/wait"
+	subscriberApi "github.com/redhat-cne/sdk-go/v1/subscriber"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	amqpLinkCredit uint32 = 50
```
Comment (jzding) [inline pkg/protocol/http/http.go:29]: amqpLinkCredit is not used

Comment (aneeshkp) [reply pkg/protocol/http/http.go]: Done

File: pkg/protocol/http/http.go [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+	}
+
+// send data
+in <- &channel.DataChan{
+	Address: addr,
+	Data:    &event,
+	Status:  channel.NEW,
+	Type:    channel.EVENT,
+}
+*/
```
Comment (jzding) [inline pkg/protocol/http/http.go:267]: remove if not used

**Review by jzding — APPROVED** [review]

File: pkg/protocol/http/http.go (hunk 4 above)
Comment (aneeshkp) [reply pkg/protocol/http/http.go:267]: That's an example

**Review by jzding — APPROVED** [review]

**Pull request merged**